### PR TITLE
Set labels in workflow metadata

### PR DESCRIPTION
**Pull request by catalinii**
https://github.com/lyft/flyte/issues/521

### CODE REVIEW

File: pkg/workflowengine/impl/propeller_executor.go
Comment (katrogan) [inline pkg/workflowengine/impl/propeller_executor.go:139]: why are we duplicating the labels? isn't passing them once sufficient?

File: pkg/manager/impl/execution_manager.go
Comment (katrogan) [inline pkg/manager/impl/execution_manager.go:624]: if the err is not nil, the project definitely won't be populated. we should bubble up the err and return in this case

File: pkg/manager/impl/execution_manager.go
Comment (katrogan) [inline pkg/manager/impl/execution_manager.go:625]: you should call the transformer: https://github.com/lyft/flyteadmin/blob/eb9236eea8af44791d87c0f4113b92d0da32fa0f/pkg/repositories/transformers/project.go#L28 in order to get the deserialized label values

File: pkg/manager/impl/execution_manager.go
Comment (catalinii) [reply pkg/manager/impl/execution_manager.go:623]: addressed

**Review by katrogan — COMMENTED** [review]
looks good! mind adding to the existing unit test: https://github.com/lyft/flyteadmin/blob/master/pkg/manager/impl/execution_manager_test.go#L199 to make sure we pass the project labels as expected?

File: pkg/manager/impl/execution_manager_test.go
Comment (katrogan) [inline pkg/manager/impl/execution_manager_test.go:201]: I think you need to mock out the Get call to return a project with the proper labels

Comment (catalinii) [reply pkg/manager/impl/execution_manager_test.go]: fixed, tested and working:
Running tool: /usr/local/opt/go/libexec/bin/go test -timeout 30s github.com/lyft/flyteadmin/pkg/manager/impl -run ^TestCreateExecution$

ok  	github.com/lyft/flyteadmin/pkg/manager/impl	0.503s

**Comment by katrogan** [comment]
to fix lint can you run `make goimports` and then validate with `make lint`

File: go.mod
Comment (katrogan) [inline go.mod:29]: let's wait until the propeller changes are in (with an official release or tag) before merging this

Comment (catalinii) [reply go.mod:29]: sure

File: pkg/manager/impl/execution_manager_test.go
Comment (katrogan) [inline pkg/manager/impl/execution_manager_test.go:205]: what do you mean by this comment? I see it in the expected values below?

Comment (katrogan) [inline pkg/manager/impl/execution_manager_test.go:205]: do we want project labels to override individual labels?

Comment (catalinii) [reply pkg/manager/impl/execution_manager_test.go:205]: They do not override, see the code: https://github.com/lyft/flyteadmin/pull/129/files#diff-fc047e54b9dd82ca7c89ac9b32cb07b3R1375

Comment (katrogan) [reply pkg/manager/impl/execution_manager_test.go:205]: ah thank you for clarifying!

**Review by katrogan — APPROVED** [review]

File: pkg/manager/impl/execution_manager.go
Comment (kumare3) [inline pkg/manager/impl/execution_manager.go:1360]: I know you have a comment below, but comments on the func become docs in golang.
So a comment like
// Adds project labels with higher precedence to workflow labels. Project labels are ignored if a corresponding label is set on the workflow.

**Comment by kumare3** [comment]
thank you looks good to me, just one comment and then +1

File: pkg/manager/impl/execution_manager.go
Comment (catalinii) [reply pkg/manager/impl/execution_manager.go:1361]: @kumare3 I addressed the comment

**Review by kumare3 — APPROVED** [review]

**Pull request merged**
